test/libsolidity/syntaxTests/storageLayoutSpecifier/constant_divided_by_zero.sol

```
@@ -1,4 +1,4 @@
 uint constant N = 100;
 contract C layout at N / 0 {}
 // ----
-// TypeError 6396: (44-49): The base slot of the storage layout must evaluate to a rational number.
+// TypeError 1505: (44-49): The base slot expression contains elements that are not yet supported by the internal constant evaluator and therefore cannot be evaluated at compilation time.
```

I find this message a bit too vague and a bit too verbose. I'm pretty sure we'll never support division by 0 in the constant evaluator (or otherwise), but then, `ConstantEvaluator` returns a `nullopt` in cases where division by zero is detected, and just swallows the error message, so it's practically impossible to differentiate cases.

Eh, you know what, the message is good as as, ignore my rambling.

---

I agree. This should just say that you can't divide by zero at compilation time. We should adjust `ConstantEvaluator` to report with more granularity why it cannot be evaluated. This way we will be able to tell the user which specific element is not yet supported (type conversion, member access, non-const builtin, etc.)

For the purpose of this PR it's good enough though. We should merge it and fix that separately.